pkg/compute/deploy.go

```diff
@@ -41,137 +56,437 @@ func NewDeployCommand(parent common.Registerer, client api.HTTPClient, globals *
 	c.CmdClause.Flag("service-id", "Service ID").Short('s').StringVar(&c.manifest.Flag.ServiceID)
 	c.CmdClause.Flag("version", "Number of version to activate").Action(c.Version.Set).IntVar(&c.Version.Value)
 	c.CmdClause.Flag("path", "Path to package").Short('p').StringVar(&c.Path)
+	c.CmdClause.Flag("domain", "The name of the domain associated to the package").StringVar(&c.Domain)
+	c.CmdClause.Flag("backend", "A hostname, IPv4, or IPv6 address for the package backend").StringVar(&c.Backend)
+	c.CmdClause.Flag("backend-port", "A port number for the package backend").UintVar(&c.BackendPort)
 
 	return &c
 }
 
 // Exec implements the command interface.
 func (c *DeployCommand) Exec(in io.Reader, out io.Writer) (err error) {
-	var progress text.Progress
-	if c.Globals.Verbose() {
-		progress = text.NewVerboseProgress(out)
-	} else {
-		progress = text.NewQuietProgress(out)
+	// Exit early if no token configured.
+	_, s := c.Globals.Token()
+	if s == config.SourceUndefined {
+		return errors.ErrNoToken
```

Comment:
6. This logic was originally only in the `init` function but is moved from there to here as we no longer make API requests in `init`. I've also added this token check to the pre-existing `publish` subcommand.